### PR TITLE
ADEN-6696 Add rabbit results to targeting key-vals

### PR DESCRIPTION
**Pull request by fraszczakszymon**

### CODE REVIEW

File: extensions/wikia/AdEngine/js/ml/rabbit.js [hunk 1]
```diff
@@ -25,19 +25,30 @@ define('ext.wikia.adEngine.ml.rabbit', [
...
 
-	function getSerializedResults() {
+	function getResults(allowedModels) {
+		allowedModels = allowedModels || [];
+
 		var results = [];
 
 		models.forEach(function (model) {
 			if (model && model.isEnabled()) {
+				if (allowedModels.length > 0 && allowedModels.indexOf(model.getName()) === -1) {
```
Comment (kvas-damian) [inline extensions/wikia/AdEngine/js/ml/rabbit.js]: Do we need to check `allowedModels.length > 0`? I believe second codnition is enough. Also you could combine it with the `if` above.

File: extensions/wikia/AdEngine/js/ml/outstream/outstreamLogisticRegression.js [hunk 2]
```diff
@@ -23,6 +23,7 @@ define('ext.wikia.adEngine.ml.outstream.outstreamLogisticRegression', [
 		model: linearModel.create(coefficients, intercept),
 		name: 'outstreamlr',
 		wgCountriesVariable: 'wgAdDriverOutstreamLogisticRegressionRabbitCountries',
-		enabled: true
+		enabled: true,
+		cachePrediction: true
```
Comment (kvas-damian) [inline extensions/wikia/AdEngine/js/ml/outstream/outstreamLogisticRegression.js:27]: Is is ok, that we cache the prediction here? I'm worried about mobile-wiki and consecutive pageviews. Is the state cleaned on every pageview?

File: extensions/wikia/AdEngine/js/ml/rabbit.js (hunk 1 above)
Comment (fraszczakszymon) [reply extensions/wikia/AdEngine/js/ml/rabbit.js]: `getSerializedResults` get predictions from all enabled models
https://github.com/Wikia/app/pull/14764/files#diff-8980c162209bf6e6a09200fc34a7b261R47

File: extensions/wikia/AdEngine/js/ml/outstream/outstreamLogisticRegression.js (hunk 2 above)
Comment (fraszczakszymon) [reply extensions/wikia/AdEngine/js/ml/outstream/outstreamLogisticRegression.js:27]: :gem: thanks for catching this. Addressed

File: extensions/wikia/AdEngine/js/ml/rabbit.js [hunk 3]
```diff
@@ -25,19 +25,30 @@ define('ext.wikia.adEngine.ml.rabbit', [
...
 				results.push(model.getResult());
 			}
 		});
 
-		return results.join(';');
+		return results;
+	}
+
+	function getAllSerializedResults() {
+		var allowedModels = models.map(function (model) {
```
Comment (kvas-damian) [inline extensions/wikia/AdEngine/js/ml/rabbit.js]: Can we extract it to a separate method? Maybe `getModelNames`

**Review by kvas-damian — APPROVED** [review]

**Review by kvas-damian — APPROVED** [review]

**Pull request merged**